dbNSFP.pm

```diff
@@ -386,28 +402,28 @@ sub run {
     %$data = %$tmp_data;
 
     # convert data with multiple transcript values
-    # if($data->{Ensembl_transcriptid} =~ m/\;/) {
-
-    #   # find the "index" of this transcript
-    #   my @tr_ids = split(';', $data->{Ensembl_transcriptid});
-    #   my $tr_index;
-
-    #   for my $i(0..$#tr_ids) {
-    #     $tr_index = $i;
-    #     last if $tr_ids[$tr_index] =~ /^$tr_id(\.\d+)?$/;
-    #   }
-
-    #   next unless defined($tr_index);
-
-    #   # now alter other fields
-    #   foreach my $key(keys %$data) {
-    #     if($data->{$key} =~ m/\;/) {
-    #       my @split = split(';', $data->{$key});
-    #       die("ERROR: Transcript index out of range") if $tr_index > $#split;
-    #       $data->{$key} = $split[$tr_index];
-    #     } 
-    #   }
-    # }
+    if($self->{transcript_match} && $data->{Ensembl_transcriptid} =~ m/\;/) {
+
+      # find the "index" of this transcript
+      my @tr_ids = split(';', $data->{Ensembl_transcriptid});
+      my $tr_index;
+
+      for my $i(0..$#tr_ids) {
+        $tr_index = $i;
+        last if $tr_ids[$tr_index] =~ /^$tr_id(\.\d+)?$/;
```

Comment:
I suggest comparing the regexp match with a simple eq test:
if ($tr_ids[$tr_index] eq $tr_id)
AFAICT, ENSTs in Ensembl_transcriptid are unversioned. So the result should be identical, but speed could be much better (no regexp compiling)